src/container/Movies.js

```diff
@@ -28,18 +67,70 @@ handleDelete(e){
 
 }
 	componentDidMount(){
+		document.title = "vaibhav app";
+		this.setState({isloading: true})
 		axios.get(`http://localhost:3001/api/v1/movies.json`)
 		.then(res => {
 			console.log(res)
-			this.setState({movies_list:res.data})
+			this.setState({
+				movies_list:res.data,
+				isloading: false})
 		} )
 
 	}
+
+	filter(criteria){
+			var search_list;
+      let name= criteria.target.name
+      let value =criteria.target.checked
+      console.log(name)
+      if(value == true){
+    		var search_list = this.state.movies_list.filter(
+			(movie)=> (
+				movie.rating >= name
+			));
+		this.setState({
+			movies_list : search_list
+		});  	
+      }else{
+      	this.componentDidMount()
```

Comment:
We can't call component life-cycle methods. They are automatically called while rendering a component. Modify the code accordingly.